spec/controllers/api/v1/animals_controller_spec.rb

```diff
@@ -38,14 +38,38 @@
       expect(response.status).to eq 200
       expect(response.content_type).to eq("application/json")
 
-      expect(returned_json.length).to eq 1
+      expect(returned_json.length).to eq 2
 
       expect(returned_json["animals"][0]["name"]).to eq "Shannon"
       expect(returned_json["animals"][0]["species"]).to eq "Chicken"
 
       expect(returned_json["animals"][1]["name"]).to eq "Jance"
       expect(returned_json["animals"][1]["species"]).to eq "Bird"
     end
+
+    it "should send user_role as admin if signed in user admin" do
+      user = FactoryBot.create(:user)
+      user.role = "admin"
+      user.save
+      sign_in user
+      get :index
+      returned_json = JSON.parse(response.body)
+      expect(returned_json["user_role"]).to eq "admin"
+    end
+
+    it "should send user_role as user if signed in not admin" do
+      user = FactoryBot.create(:user)
+      sign_in user
+      get :index
+      returned_json = JSON.parse(response.body)
+      expect(returned_json["user_role"]).to eq "user"
+    end
```

Comment:
if this would already be returned in a normal index request, add this to your original animal index controller spec